open-craft/opencraft
# [BB-835] Use a sub-domain instead of a prefixed domain for Studio, Preview etc.

**Description**
From the Ironwood release, it is a [requirement](https://groups.google.com/d/msg/openedx-ops/521im9B92Xo/nS2aLWOGHwAJ) to serve the LMS and Studio from cookie-compatible domains. So the studio domain is now a sub-domain of the LMS domain. To allow session sharing, the `EDXAPP_SESSION_COOKIE_DOMAIN` ansible variable has been set to `.<LMS domain>`. For consistency, all the other domains - preview, discovery, ecommerce are updated to be sub-domains of the LMS domain.

This also adds a new ansible variable `EDXAPP_LOGIN_REDIRECT_WHITELIST` which allows the redirect to Studio after logging in via the LMS.

**Testing instructions**
1. Run Ocim with the PR branch.
1. Apply the migrations.
1. Create a new instance. Verify that the internal domains for Studio, Preview, Discovery and E-Commerce are now sub-domains of the internal LMS domain and that the `enable_prefix_domains_redirect` field is set to `False`.
1. Launch a new appserver for the instance and activate it after the provisioning completes successfully.
1. Verify that the load-balancer configuration is added correctly for the internal domains and that there are no redirects setup for the instance.
1. Add external domains for the same instance and launch a new appserver. Verify that the configuration is added correctly for the internal and external domains and that there are no redirects setup for the instance.
1. Create a new instance and set the `enable_prefix_domains_redirect` field to `True`.
1. Launch a new appserver for the instance and activate it after the provisioning completes successfully.
1. Verify that the load-balancer configuration is added correctly for the internal domains and that there are redirects from the prefix domains to the corresponding sub-domains. Verify that the DNS records have been added for the prefix domains and that the redirect works as expected.
1. Add external domains for the same instance and launch a new appserver. Verify that the configuration is added correctly for the internal and external domains and that the redirects are setup only for the prefix domains to the corresponding internal sub-domains. Verify that the internal domains, prefix domains and the external domains work as expected.
1. Optionally, repeat the same steps for the previous instance on an existing instance after updating its internal domains to be sub-domains.
1. Archive the instance and verify that the DNS records for the prefix domains are deleted.
1. Optionally, verify that these changes do not impact the existing hawthorn instances.

**Reviewers**
